src/Source.ml

```
@@ -52,11 +52,11 @@ let tokens_at t ?(filter= fun _ -> true) (l: Location.t) :
   loop []
 
 let loc_between ~(from: Location.t) ~(upto: Location.t) : Location.t =
-  {from with loc_end= upto.loc_start}
+  {from with loc_start= from.loc_end; loc_end= upto.loc_start}
 
 let tokens_between t ?(filter= fun _ -> true) ~(from: Location.t)
     ~(upto: Location.t) : (Parser.token * Location.t) list =
-  tokens_at t ~filter (loc_between ~from ~upto)
+  try tokens_at t ~filter (loc_between ~from ~upto) with _ -> []
```

Please no catch-all exception handlers. I've had bad days where things like Stack_overflow were getting caught.